content/en/blog/_posts/2022-11-28-kubelet-credential-providers/index.md

```diff
@@ -0,0 +1,128 @@
+
+Kubernetes v1.20 introduced alpha support for kubelet credential providers plugins,
+which provide a mechanism for the kubelet to dynamically authenticate and pull images
+for arbitrary container registries - whether these are public registries, managed services,
+or even a self-hosted registry.
+In Kubernetes v1.26, this feature is now GA
+
+{{< figure src="kubelet-credential-providers-plugin.png" caption="Figure 2: Kubelet credential provider overview" >}}
+
+## Why is it important?
+
+Prior to Kubernetes v1.20,  if you wanted to dynamically fetch credentials for image registries
+other than ACR (Azure Container Registry), ECR (Elastic Container Registry), or GCR
+(Google Container Registry), you needed to modify the kubelet code.
+The new plugin mechanism can be used in any cluster, and lets you authenticate to new registries without
+any changes to Kubernetes itself. Any cloud provider or vendor can publish a plugin that lets you authenticate with their image registry.
+
+## How it Works
+
+The kubelet and the credential provider plugin binary communicate through stdio (stdin, stdout, and stderr) by sending and receiving
+json-serialized api-versioned types. If the exec plugin is enabled and the kubelet requires authentication information for an image
+that matches against a plugin, the kubelet will exec the plugin binary, passing the `CredentialProviderRequest` API via stdin. Then
+the exec plugin communicates with the container registry to dynamically fetch the credentials and returns the credentials in an
+encoded response of the `CredentialProviderResponse` API to the kubelet via stdout.
+
+{{< figure src="kubelet-credential-providers-how-it-works.png" caption="Figure 3: Kubelet credential provider plugin flow" >}}
+
+On receiving credentials from the Kubelet, the plugin can also indicate how long credentials can be cached for, to prevent unecessary
+execution of the plugin by the Kubelet for subsequent image pull requests to the same registry. In cases where the cache duration
+is not specified by the plugin, a default cache duration can be specified by the kubelet (more details below).
+
+```json
+{
+  "apiVersion": "kubelet.k8s.io/v1",
+  "kind": "CredentialProviderResponse",
+  "auth": {
+    "cacheDuration": "6h",
+    "private-registry.io/my-app": {
+      "username": "“user”",
+      "password": "“token12345”"
+    }
+  }
+}
+```
+
+In addition, the plugin can specify the scope in which cached credentials are valid for. This is specified through the `cacheKeyType` field
+in `CredentialProviderResponse`. When the value is `Image`, the kubelet will only use cached credentials for future image pulls that exactly
+match the image of the first request. When the value is `Registry`, the kubelet will use cached credentials for any subsequent image pulls
```

Comment:
For the Registry value, do we mean to say that the kubelet uses cached credentials for image pulls from the same registry host? Adding the verb originate here could confuse the audience because they might assume that the image-pulling action originates at the registry host end.
```suggestion
match the image of the first request. When the value is `Registry`, the kubelet will use cached credentials for any subsequent image pulls
```